riot.d.ts

```diff
@@ -1,71 +1,71 @@
 // This interface is only exposed and any Riot component will receive the following properties
-export interface RiotCoreComponent {
+export interface RiotCoreComponent<P = object, S = object> {
   // automatically generated on any component instance
-  readonly props: object
+  readonly props: P
   readonly root: HTMLElement
   readonly name?: string
   // TODO: add the @riotjs/dom-bindings types
   readonly slots: any[]
   mount(
     element: HTMLElement,
-    initialState?: object,
+    initialState?: S,
     parentScope?: object
-  ): RiotComponent
+  ): RiotComponent<P, S>
   update(
-    newState?: object,
+    newState?: Partial<S>,
     parentScope?: object
-  ): RiotComponent
-  unmount(keepRootElement: boolean): RiotComponent
+  ): RiotComponent<P, S>
+  unmount(keepRootElement: boolean): RiotComponent<P, S>
 
   // Helpers
   $(selector: string): HTMLElement
   $$(selector: string): [HTMLElement]
 }
 
 // This object interface is created anytime a riot file will be compiled into javascript
-export interface RiotComponentShell {
+export interface RiotComponentShell<P = object, S = object> {
   readonly css?: string
-  readonly exports?: () => RiotComponent|object
+  readonly exports?: () => RiotComponent<P, S>|object
```

Comment:
here we should need to change `RiotComponent` -> `RiotComponentExport`

Reply:
Now fixed. Thank you.